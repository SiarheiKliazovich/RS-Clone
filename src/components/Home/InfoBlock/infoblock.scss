@import "../../../sass/variables";
@import "../../../sass/mixins";

.section-space {
  background:url(../../../assets/images/png/06-TIE-STORE-BANNER-MIXED-1920-6.png);
  padding: 100px 0;
  border-top: 3px solid $color-white1;
  border-bottom: 3px solid $color-white1;
  background-size: cover;
}

.section-space__banner {
  background: $color-grey8;
  padding: 30px 0;
}

.row-points-wrapper {
  display: flex;
  flex-wrap: wrap;

  @include media-mobile {
    display: flex;
    flex-direction: column;
  }
}
  
.points-wrapper {
  width: 33%;
  text-align: center;
  background:$color-white3;
  z-index: 1;
  padding: 15px 15px 15px 15px;
  
  
  @include media-mobile {
    width: 100%;
    padding-bottom: 15px;
  }
}

.custom-tab {
  margin-right: 30px;

  @include media-mobile {
    margin-right: 20px;
    margin-left: 20px;
  }
}

.custom-tab__nav_list {
  border-radius: 0;
  width: 100%;
  -ms-box-shadow: 0 22px 40px 0 $color-black3;
  -o-box-shadow: 0 22px 40px 0 $color-black3;
  box-shadow: 0 22px 40px 0 $color-black3;
  background: transparent;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  padding-left: 0;
  margin-bottom: 0;
  list-style: none;
  border-radius: 5px;
}

.custom-tab__item {
  border: 1px solid $color-white1;
  width: 100%;
  vertical-align: middle;
  float: left;
  cursor: pointer;
  z-index: 10;

  &:hover {
    opacity: 0.9;
  }
}

.select {
  color: $color-white1;
}

.tab-title {
  font-size: $font-size-llg;
  padding: 10px 0px;
  font-family: $font-family-base;
  cursor: pointer;

  &:hover {
    opacity: 0.7;
  }

  @include media-mobile {
    font-size: $font-size-llg;
  }
}

.points-info-block {
  width: 66%;
  background:$color-white3;
  z-index: 1;
  padding: 18px 19px 1px 0px;

  @include media-mobile {
    width: 100%;
  }
}

.tab-pane-content {
  padding-top: 40px;
}

.tab-pane-content {
  @include media-mobile {
    padding-left: 15px;
  }
}

.tab-pane__title {
  font-size: $font-size-xlg;
  font-family: $font-family-base;
  padding-bottom: 15px;
  text-align: center;
}

.tab-pane__text {
  font-family: $font-family-base;
  font-size: $font-size-lgg;
  font-weight: $font-weight-normal;
  margin-bottom: 10px;
  color:$color-black1;
}

.tab-pane__list {
  columns: 2;
  padding-top: 10px;
  padding-bottom: 10px;

  @include media-mobile {
    columns: 1;
  }
}

.tab-pane__list_item {
  padding-left: 30px;
  margin-bottom: 20px;
  position: relative;
  font-family: $font-family-base;
  font-size: $font-size-llg;
}

