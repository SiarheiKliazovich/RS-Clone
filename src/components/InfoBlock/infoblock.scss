@import "../../sass/variables";
@import "../../sass/mixins";

.section-space {
  background:url(../../assets/images/png/06-TIE-STORE-BANNER-MIXED-1920-6.png);
  padding: 100px 0;
  border-top: 3px solid #ffffff;
  border-bottom: 3px solid #ffffff;
}

.section-space__banner {
  background: #eff1fe8c;
  padding: 30px 0;
}

.container-section {
  width: 98%;
  padding-right: 15px;
  padding-left: 15px;
  margin-right: auto;
  margin-left: auto;
}

.row-points-wrapper {
  display: flex;
  flex-wrap: wrap;
}

.points-wrapper {
  width: 33%;
  text-align: center;
  background:#eff1fee6;
  z-index: 1;
  padding-top: 15px;
}

.custom-tab {
  margin-right: 30px;
}

.custom-tab__nav_list {
  border-radius: 0;
  width: 100%;
  -ms-box-shadow: 0 22px 40px 0 rgba(0,0,0,.05);
  -o-box-shadow: 0 22px 40px 0 rgba(0,0,0,.05);
  box-shadow: 0 22px 40px 0 rgb(0 0 0 / 5%);
  background: transparent;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  padding-left: 0;
  margin-bottom: 0;
  list-style: none;
}

.custom-tab__item {
  border: 1px solid #fff;
  width: 100%;
  vertical-align: middle;
  float: left;
  cursor: pointer;
  z-index: 10;

  &:hover {
    background: #4788ff;
   
  }
}

.active {
  background: #4788ff;
  color: #fff;
}

.tab-title {
  font-size: $font-size-xlg;
  padding: 10px 0px;
  font-family: $font-family-base;

  &:hover { 
    color: #fff;
  }
}

.points-info-block {
  width: 66%;
  background:#eff1fee6;
  z-index: 1;
}

.tab-pane__title {
  font-size: $font-size-xlg;
  font-family: $font-family-base;
  padding-bottom: 15px;
}

.tab-pane__text {
  font-family: $font-family-base;
  font-size: $font-size-llg;
  font-weight: $font-weight-normal;
  margin-bottom: 10px
}

.tab-pane__list {
  columns: 2;
  padding-top: 10px;
  padding-bottom: 10px;
}

.tab-pane__list_item {
  padding-left: 30px;
  margin-bottom: 20px;
  position: relative;
  font-family: $font-family-base;
  font-size: $font-size-llg;
}

.check-circle {
  color: #4788ff;
  left: 0;
  position: absolute;
  
  &::before {
    content: url(../../assets/images/png/check-mark.png);
  }
}