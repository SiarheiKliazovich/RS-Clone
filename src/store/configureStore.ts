/* eslint-disable @typescript-eslint/ban-ts-comment */
import { applyMiddleware, createStore } from "redux";
import thunkMiddleware from "redux-thunk";

import loggerMiddleware from "./logger";
import rootReducer from "../reducers";

const configureStore = () => {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(rootReducer, middlewareEnhancer);

  // @ts-ignore:next-line
  if (process.env.NODE_ENV !== "production" && module.hot) {
    // @ts-ignore:next-line
    module.hot.accept("./../reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
};

export default configureStore;
