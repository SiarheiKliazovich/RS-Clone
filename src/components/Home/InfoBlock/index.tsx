import { FunctionComponent, useState } from "react";
import { useSelector } from "react-redux";
import "./infoblock.scss";
import classNames from "classnames";
import { nightTheme } from "../../../data/constants";
import { InfoBlockType } from "../../../types/home";
import infoSection from "../../../data/infoSection";
import { ILangReducer } from "../../../interfaces/langReducer";

const InfoBlock: FunctionComponent<InfoBlockType> = ({
  accentColor,
  isNavbarNightMode,
}: InfoBlockType) => {
  const [tab, setTab] = useState<number>(1);
  const backgroundColor = nightTheme.background.element;
  const { lang } = useSelector((state: ILangReducer) => state.langReducer);
  const list = infoSection.find((c) => c.lang === lang)!;

  return (
    <section className="section-space">
      <div className="section-space__banner">
        <div className="container">
          <div className="row-points-wrapper">
            <div className="points-wrapper">
              <div className="custom-tab">
                <ul className="custom-tab__nav_list">
                  {list.data.map((item, index) => (
                    <li className="custom-tab__item" key={item.question}>
                      <h5
                        className={classNames("tab-title", {
                          "tab-title-select": tab === index + 1,
                        })}
                        style={{
                          backgroundColor: isNavbarNightMode
                            ? backgroundColor
                            : tab === index + 1
                            ? accentColor.static
                            : "",
                        }}
                        onClick={() => setTab(index + 1)}
                      >
                        {item.question}
                      </h5>
                    </li>
                  ))}
                </ul>
              </div>
            </div>
            <div className="points-info-block">
              {tab === 1 && (
                <div className="tab-pane-content">
                  {list.data[0].answer.map((answer) => (
                    <p className="tab-pane__text" key={answer}>
                      {answer}
                    </p>
                  ))}
                </div>
              )}
              {tab === 2 && (
                <div className="tab-pane-content">
                  {list.data[1].answer.map((answer) => (
                    <p className="tab-pane__text" key={answer}>
                      {answer}
                    </p>
                  ))}
                </div>
              )}
              {tab === 3 && (
                <div className="tab-pane-content">
                  {list.data[2].answer.map((answer) => (
                    <p className="tab-pane__text" key={answer}>
                      {answer}
                    </p>
                  ))}
                </div>
              )}
              {tab === 4 && (
                <div className="tab-pane-content">
                  {list.data[3].answer.map((answer) => (
                    <p className="tab-pane__text" key={answer}>
                      {answer}
                    </p>
                  ))}
                </div>
              )}
            </div>
          </div>
        </div>
      </div>
    </section>
  );
};
export default InfoBlock;
