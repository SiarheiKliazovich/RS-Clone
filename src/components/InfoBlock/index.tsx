import { FunctionComponent, useEffect, useState } from "react";
import "./infoblock.scss";
import classNames from "classnames";

const InfoBlock: FunctionComponent = () => {
  const [pointActive, setPointActive] = useState<boolean>(false);
  return (
    <section className="section-space">
      <div className="section-space__banner">
        <div className="container-section">
          <div className="row-points-wrapper">
            <div className="points-wrapper">
              <div className="custom-tab">
                <ul className="custom-tab__nav_list">
                  <li
                    className="custom-tab__item "
                    // onClick={() => setPointActive(pointActive)}
                  >
                    <a>
                      <h5
                        className={classNames("tab-title", {
                          active:
                            pointActive === true
                              ? !pointActive
                              : pointActive === false,
                        })}
                        onClick={() => setPointActive(true)}
                      >
                        Blok Information 1
                      </h5>
                    </a>
                  </li>
                  <li
                    className="custom-tab__item"
                    // onClick={() => setPointActive(pointActive)}
                  >
                    <a>
                      <h5
                        className={classNames("tab-title ", {
                          active:
                            pointActive === true
                              ? !pointActive
                              : pointActive === false,
                        })}
                        onClick={() => setPointActive(true)}
                      >
                        Blok Information 2
                      </h5>
                    </a>
                  </li>
                  <li
                    className="custom-tab__item"
                    // onClick={() => setPointActive(pointActive)}
                  >
                    <a>
                      <h5
                        className={classNames("tab-title ", {
                          active:
                            pointActive === true
                              ? !pointActive
                              : pointActive === false,
                        })}
                        onClick={() => setPointActive(true)}
                      >
                        Blok Information 3
                      </h5>
                    </a>
                  </li>
                  <li
                    className="custom-tab__item"
                    // onClick={() => setPointActive(pointActive)}
                  >
                    <a>
                      <h5
                        className={classNames("tab-title ", {
                          active:
                            pointActive === true
                              ? !pointActive
                              : pointActive === false,
                        })}
                        onClick={() => setPointActive(true)}
                      >
                        Blok Information 4
                      </h5>
                    </a>
                  </li>
                </ul>
              </div>
            </div>
            <div className="points-info-block">
              <div className="tab-pane">
                <div className="tab-pane-content">
                  <h3 className="tab-pane__title">Title of information</h3>
                  <p className="tab-pane__text">
                    Our Ties is very best.Choose us because:
                  </p>
                  <ul className="tab-pane__list">
                    <li className="tab-pane__list_item">
                      <i className="check-circle"></i>
                      Advance Advisory Team
                    </li>
                    <li className="tab-pane__list_item">
                      <i className="check-circle"></i>
                      Professional Consulting Services
                    </li>
                    <li className="tab-pane__list_item">
                      <i className="check-circle"></i>
                      24/7 Support Help Center
                    </li>
                    <li className="tab-pane__list_item">
                      <i className="check-circle"></i>
                      Customer Service &amp; Operations
                    </li>
                    <li className="tab-pane__list_item">
                      <i className="check-circle"></i>
                      There are many variations of passages.
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  );
};
export default InfoBlock;
