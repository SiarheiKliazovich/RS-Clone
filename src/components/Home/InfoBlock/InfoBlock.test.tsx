import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import InfoBlock from ".";

const mockStore = configureStore([]);

describe("InfoBlock", () => {
  // eslint-disable-next-line
  let store: any;

  beforeEach(() => {
    store = mockStore({
      langReducer: {
        lang: "en",
      },
    });
  });

  test("renders correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <InfoBlock
            accentColor={{ static: "#fff", hover: "#000" }}
            isNavbarNightMode={false}
          />
        </Router>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
