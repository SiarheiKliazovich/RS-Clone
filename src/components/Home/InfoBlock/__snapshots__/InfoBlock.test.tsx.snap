// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBlock renders correctly 1`] = `
<section
  class="section-space"
>
  <div
    class="section-space__banner"
  >
    <div
      class="container"
    >
      <div
        class="row-points-wrapper"
      >
        <div
          class="points-wrapper"
        >
          <div
            class="custom-tab"
          >
            <ul
              class="custom-tab__nav_list"
            >
              <li
                class="custom-tab__item"
              >
                <h5
                  class="tab-title tab-title-select"
                  style="background-color: rgb(255, 255, 255);"
                >
                  CUSTOMISED PRODUCTS
                </h5>
              </li>
              <li
                class="custom-tab__item"
              >
                <h5
                  class="tab-title"
                >
                  CAN I ORDER A READY-MADE PRODUCT OFF THE SHELF?
                </h5>
              </li>
              <li
                class="custom-tab__item"
              >
                <h5
                  class="tab-title"
                >
                  HOW ARE YOU ABLE TO OFFER QUANTITIES FROM ONE AT SUCH AFFORDABLE PRICES?
                </h5>
              </li>
              <li
                class="custom-tab__item"
              >
                <h5
                  class="tab-title"
                >
                  HOW LONG DO I HAVE TO WAIT FOR MY CUSTOM PRODUCTS?
                </h5>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="points-info-block"
        >
          <div
            class="tab-pane-content"
          >
            <p
              class="tab-pane__text"
            >
              A custom product simply means that you design your tie, novelty tie or scarf just as you want, no compromises.
            </p>
            <p
              class="tab-pane__text"
            >
              Choose a canvas, apply colors, add logos and then select your size. The limitless options allow you to create designs as individual as your company brand, values, event or even your personality. We specialize in woven and printed ties, novelty ties, and ladies neck and head-scarves. Even apply your unique message onto your tie or scarf sleeve.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
